src/offline/cable_checks.F90

```diff
@@ -50,141 +52,561 @@ MODULE cable_checks_module
+          meth = [0.0, 1.0],                 &
+          za =[0.0, 150.0],                  &
+          rpcoef = [0.05, 1.5],              &
+          shelrb = [1.0, 3.0],               &
+          vcmax = [5.0E-6, 1.5E-4],          &
+          xfang = [-1.0, 0.5],               &
+          ratecp = [0.01, 3.0],              &
+          ratecs = [0.01, 3.0],              &
+          refsbare = [0.0, 0.5],             &
+          taul = [0.0, 0.3],                 &
+          refl = [0.0, 0.5],                 &
+          tauw = [0.0, 0.1],                 &
+          refw = [0.0, 0.5],                 &
+          extkn = [0.0, 10.0],               & ! YP oct07
+          wai = [0.0, 5.0],                  & ! YP oct07
+          vegcf = [0.0, 100.0],              & ! YP oct07
+          tminvj = [-20.0, 15.0],            &
+          tmaxvj = [-15.0, 30.0],            &
+          rootbeta = [0.7, 1.0],             & ! YP oct07
+          veg_class = [1.0, 20.0],           &
+          soil_class = [1.0, 20.0]  , &
+          TotLivBiomass =  [0.0, 1000.],      &
+          TotSoilCarb =  [0.0, 1000.],      &
+          TotLittCarb =  [0.0, 1000.], &
+          Area = [0.0, 5000.],&
                                 !MD
-          WatTable = (/0.0,1.0e10/),          &
-          GWwb = (/0.0,1.0/),              &
-          SatFrac = (/0.0,1.0/),              &
-          Qrecharge = (/-9999.0,9999.0/)
+          WatTable = [0.0, 1.0e10],          &
+          GWwb = [0.0, 1.0],              &
+          SatFrac = [0.0, 1.0],              &
+          Qrecharge = [-9999.0, 9999.0], &
+          patchfrac = [0.0, 1.0], &
+
+          ! soil
+          slope = [0.0, 1.0], &
+          slope_std = [0.0, 1.0], &
+          GWdz = [0.0, 1.0], &
+
+          ! ssnow
+          ssdn = [0.0, 9999.0], &
+          smass = [0.0, 9999.0], &
+          sdepth = [0.0, 9999.0], &
+          tggsn = [100.0, 300.0], &
+          albsoiln = [0.0, 1.0], &
+          S = [0.0, 1.5], &
+          Tsoil = [-100.0, 100.0], &
+
+          ! gw model
+          gw_default = [0.0, 100000000.0], &
+
+          default_l = [-99999.0, 9999999.0], & ! default large range
+          default_s = [-99999.9, 99999.0] ! default smaller range
+
+
   END TYPE ranges_type
   TYPE(ranges_type),SAVE :: ranges
 
+  INTERFACE check_range
+      MODULE PROCEDURE :: check_range_i1_d1
+      MODULE PROCEDURE :: check_range_r1_d1
+      MODULE PROCEDURE :: check_range_r2_d1
+      MODULE PROCEDURE :: check_range_r1_d2
+      MODULE PROCEDURE :: check_range_r2_d2
+      MODULE PROCEDURE :: check_range_r1_d3
+      MODULE PROCEDURE :: check_range_r2_d3
+  END INTERFACE check_range
+
 CONTAINS
 
+  ! Make for integers as well: Example isoilm
+  SUBROUTINE check_range_i1_d1(vname, parameter_i1, parameter_range, ktau, met)
+
+     CHARACTER(LEN=*) :: vname
+     INTEGER, DIMENSION(:), INTENT(IN) :: parameter_i1
+     REAL, DIMENSION(2), INTENT(IN) :: parameter_range
```

Comment:
```suggestion
     INTEGER, INTENT(IN) :: parameter_i1(:)
     REAL, INTENT(IN) :: parameter_range(2)
```

Reply:
This came in before I was finished. I changed this because the UM/JULES coding standard asks not to use the DIMENSION keyword.